src/Processors/Formats/Impl/Parquet/ParquetBloomFilterCondition.cpp

```diff
@@ -0,0 +1,295 @@
+#include <Processors/Formats/Impl/Parquet/ParquetBloomFilterCondition.h>
+
+#if USE_PARQUET
+
+#include <parquet/bloom_filter.h>
+
+#include <DataTypes/DataTypeTuple.h>
+#include <DataTypes/DataTypeArray.h>
+#include <DataTypes/DataTypeNullable.h>
+#include <DataTypes/DataTypeLowCardinality.h>
+#include <Interpreters/misc.h>
+#include <Interpreters/convertFieldToType.h>
+#include <Columns/ColumnConst.h>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int LOGICAL_ERROR;
+    extern const int BAD_ARGUMENTS;
+}
+
+namespace
+{
+    parquet::ByteArray createByteArray(std::string_view view, TypeIndex type, uint8_t * buffer, uint32_t buffer_size)
+    {
+        if (isStringOrFixedString(type))
+        {
+            return view;
+        }
+        else
+        {
+            auto size = static_cast<uint32_t>(std::max(view.size(), sizeof(uint32_t)));
+            chassert(size <= buffer_size);
+            std::copy(view.begin(), view.end(), buffer);
+            return parquet::ByteArray(size, buffer);
+        }
+    }
+
+    ColumnPtr hash(const IColumn * data_column, const std::unique_ptr<parquet::BloomFilter> & bloom_filter)
+    {
+        static constexpr uint32_t buffer_size = 32;
+        uint8_t buffer[buffer_size] = {0};
+
+        auto column_size = data_column->size();
+
+        auto hashes_column = ColumnUInt64::create(column_size);
+        ColumnUInt64::Container & hashes_internal_data = hashes_column->getData();
+
+        for (size_t i = 0; i < column_size; ++i)
+        {
+            const auto data_view = data_column->getDataAt(i).toView();
+
+            const auto ba = createByteArray(data_view, data_column->getDataType(), buffer, buffer_size);
+
+            const auto hash = bloom_filter->Hash(&ba);
```

Comment:
This doesn't need bloom filter, can directly use `parquet::XxHasher` from `parquet/xxhasher.h`. No need to read bloom filter for first row group - it's wasteful do actual IO from disk or network then not use any of the retrieved data.